### PR TITLE
Javascript -> Typescript

### PR DESCRIPTION
**Pull request by heeckhau**
## Description

Converted Javascript files to Typescript

## Related Issue

#6392

### CODE REVIEW

File: src/components/UpcomingEventsList.tsx [hunk 1]
```diff
@@ -3,15 +3,27 @@ import React, { useEffect, useState } from "react"
...
+import EventCard from "./EventCard"
+import InfoBanner from "./InfoBanner"
+import Link from "./Link"
+import Translation from "./Translation"
+import ButtonLink from "./ButtonLink"
 
 // Data
 import events from "../data/community-events.json"
 
+type EventType = {
```
Comment (pettinarip) [inline src/components/UpcomingEventsList.tsx:15]: Lets break this out and refactor a bit the name.

```ts
interface RawCommunityEvent {
  title: string
  to: string
  sponsor: string | null
  location: string
  description: string
  startDate: string
  endDate: string
}

interface CommunityEvent extends RawCommunityEvent {
  date: string
  formattedDetails: string
}
```

File: src/components/UpcomingEventsList.tsx [hunk 2]
```diff
@@ -50,20 +62,28 @@ const ButtonLinkContainer = styled.div`
...
+    return new Date(
+      parseInt(parts[0]),
+      parseInt(parts[1]) - 1,
+      parseInt(parts[2])
+    )
   }
 
   useEffect(() => {
-    const eventsList = [...events]
+    const eventsList: EventType[] = [...events] as EventType[]
```
Comment (pettinarip) [inline src/components/UpcomingEventsList.tsx:86]:
```suggestion
    const eventsList: Array<RawCommunityEvent> = [...events]
```

File: src/pages/contributing/translation-program/acknowledgements.tsx [hunk 3]
```diff
@@ -265,7 +272,7 @@ const TranslatorAcknowledgements = ({ data, location }) => {
 export default TranslatorAcknowledgements
 
 export const query = graphql`
-  query {
+  query TranslatorAcknowledgements {
```
Comment (pettinarip) [inline src/pages/contributing/translation-program/acknowledgements.tsx:275]:
```suggestion
  query TranslatorAcknowledgementsPage {
```

**Comment by heeckhau** [comment]
@pettinarip Thanks for the feedback. I have rebased and applied the requested changes.

I stil have one issue that I don't know how to handle: In `deposit-contract.tsx` I get an error message on the `mt` property: https://github.com/heeckhau/ethereum-org-website/blob/typescript/src/pages/staking/deposit-contract.tsx#L472

```
src/pages/staking/deposit-contract.tsx:472:62 - error TS2322: Type '{ children: Element; isWarning: true; emoji: string; mt: string; }' is not assignable to type 'IntrinsicAttributes & IProps & { children?: ReactNode; }'.
  Property 'mt' does not exist on type 'IntrinsicAttributes & IProps & { children?: ReactNode; }'.

472               <InfoBanner isWarning={true} emoji=":warning:" mt={`2rem`}>
```

Can you take a look?

**Review by pettinarip — APPROVED** [review]
GJ @heeckhau! thanks for all this work.

**Pull request merged**